File: 0989-add-to-array-form-of-integer/0989-add-to-array-form-of-integer.java
class Solution {
    public List<Integer> addToArrayForm(int[] num, int k) {
        ArrayList<Integer> list = new ArrayList<>();
        int i = num.length-1;
        while(i >= 0 || k >0){
            if(i>=0){
                list.add((num[i] + k)%10);
                k = (num[i] +k)/10;
            }else{
                list.add(k%10);
                k/=10;
            }
            i--;
        }
        Collections.reverse(list);
        return list;
        
    }
}

File: 0989-add-to-array-form-of-integer/0989-add-to-array-form-of-integer.js
/**
 * @param {number[]} num
 * @param {number} k
 * @return {number[]}
 */
var addToArrayForm = function(num, k) {
    let arr = [];
    let i = num.length-1;
    while(i>=0 || k>0){
        if(i>=0){
            k+=num[i];
        }
        arr.push(k%10);
        k = Math.floor(k/10);
        i--;
    }
    arr.reverse();
    return arr;
};